src/main/java/org/bih/eos/converter/cdt/converter/configurable/specific/MeasurementConverter.java

```diff
@@ -114,21 +115,29 @@ protected MeasurementEntity convertValuePath(Locatable contentItem, String path,
     }
 
     private MeasurementEntity convertDataValue(MeasurementEntity measurement, Element element, DVTextCodeToConceptConverter conceptConverter) {
-        if (element.getValue() != null) {
-            if (element.getValue().getClass().equals(DvQuantity.class)) {
-                measurement.setValue(DvGetter.getDvQuantity((DvQuantity) element.getValue()));
-            } else if (element.getValue().getClass().equals(DvText.class)) {
-                measurement.setValue(DvGetter.getDvText((DvText) element.getValue()));
-            } else if (element.getValue().getClass().equals(DvCodedText.class)) {
-                measurement.setValue(conceptConverter.convertStandardConcept(element.getValue()),
-                        DvGetter.getDvCodedText((DvCodedText) element.getValue()));
-            } else if (element.getValue().getClass().equals(DvOrdinal.class)) {
-                setDvOrdinal((DvOrdinal) element.getValue(), measurement, conceptConverter);
-            } else if (element.getValue().getClass().equals(DvProportion.class)) {
-                setDvProportion((DvProportion) element.getValue(), measurement, conceptConverter);
-            }
-        }
-        return measurement;
+    	if (element.getValue() != null) {
+    		if (element.getValue().getClass().equals(DvQuantity.class)) {
+    			measurement.setValue(DvGetter.getDvQuantity((DvQuantity) element.getValue()));
+    		} else if (element.getValue().getClass().equals(DvText.class)) {
+    			Optional<String> dvText = DvGetter.getDvText((DvText) element.getValue());
```

Comment:
we need to clean up this method.
i think moving this into an own method should solve it.